maps/randomvaults/objects.dm

```diff
@@ -630,3 +636,213 @@
 
 		CO.face_atom(AM)
 		CO.visible_message("<span class='notice'>\The [CO] looks at \the [AM] and hisses angrily!</span>")
+
+
+/* Spy Sat stuff */
+
+/obj/machinery/power/magtape_deck
+	name = "magnetic tape drive"
+	desc = "A primitive way of storing information. Used because of its longevity over most digital counterparts."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "blackbox"
+
+	active_power_usage = 500
+	density = 1
+	var/last_check
+
+/obj/machinery/power/magtape_deck/New()
+	..()
+	connect_to_network()
+
+	last_check = world.time
+
+/obj/machinery/power/magtape_deck/power_change()
+	if(get_powernet() && avail(active_power_usage))
+		stat &= ~NOPOWER
+	else
+		stat |= NOPOWER
+	update_icon()
+
+	if(stat & NOPOWER)
+		return 0
```

Comment:
This returns a bool but it doesn't return anything when the proc exists successfully.

Reply:
I thought return 1 at the end of the proc would be redundant